nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -1047,6 +1047,24 @@ int ddxProcessArgument(int argc, char *argv[], int i)
     return 0;
   }
 
+  /*
+   * Disable Xinerama (i.e. fake it in Screen.c) if somehow Xinerama support
+   * has been disabled on the cmdline.
+   */
+#if defined(PANORAMIX) && defined(RANDR)
+  if ((noPanoramiXExtension || PanoramiXExtensionDisabledHack) && noRRXineramaExtension)
+    nxagentChangeOption(Xinerama, 0);
+#else
+#ifdef RANDR
+  if (noRRXineramaExtension)
+    nxagentChangeOption(Xinerama, 0);
+#endif
+#ifdef PANORAMIX
+  if (noPanoramiXExtension || PanoramiXExtensionDisabledHack)
+    nxagentChangeOption(Xinerama, 0);
+#endif
+#endif
+
   return 0;
```

Comment:
What about

```
  if (
  #if defined(PANORAMIX) && defined(RANDR)
   ((noPanoramiXExtension || PanoramiXExtensionDisabledHack) && noRRXineramaExtension)
  #else 
    #ifdef RANDR 
      (noRRXineramaExtension) 
    #else
      (noPanoramiXExtension || PanoramiXExtensionDisabledHack) 
    #endif 
  #endif
  ) {
     nxagentChangeOption(Xinerama, 0); 
  }
```

We could remove some of the left parentheses, too, but that makes it less readable again.

Or maybe this:
```
#define PANORAMIXCOND True
#define RANDRCOND True

#ifdef PANORAMIX 
  #define PANORAMIXCOND (noPanoramiXExtension || PanoramiXExtensionDisabledHack)
#endif

#ifdef RANDR 
  #define RANDRCOND noRRXineramaExtension
#endif

if (PANORAMIXCOND && RANDRCOND) 
 nxagentChangeOption(Xinerama, 0); 
```